src/compiler/utilities.ts

```diff
@@ -2744,6 +2744,20 @@ namespace ts {
     }
 
     /**
+     * Gets the effective return type annotation of a signature. If the node was parsed in a
+     * JavaScript file, gets the return type annotation from JSDoc.
+     */
+    export function getEffectiveTypeParameterDeclarations(node: DeclarationWithTypeParameters): TypeParameterDeclaration[] {

```

Comment:
No, there is no perf impact. Monaco gets 0.05s slower, TFS gets 0.04s faster on average over 20 iterations. But neither number is statistically significant.